core/src/main/scala/filodb.core/memstore/TimeSeriesShard.scala

```diff
@@ -1455,18 +1465,31 @@ class TimeSeriesShard(val ref: DatasetRef,
                        querySession: QuerySession): PartLookupResult = {
     querySession.lock = Some(reclaimLock)
     reclaimLock.lock()
+    val metricShardKeys = schemas.part.options.shardKeyColumns.dropRight(1)
     // any exceptions thrown here should be caught by a wrapped Task.
     // At the end, MultiSchemaPartitionsExec.execute releases the lock when the task is complete
     partMethod match {
       case SinglePartitionScan(partition, _) =>
         val partIds = debox.Buffer.empty[Int]
         getPartition(partition).foreach(p => partIds += p.partID)
-        PartLookupResult(shardNum, chunkMethod, partIds, Some(RecordSchema.schemaID(partition)))
+        val chunksQueriedMetric = shardStats.chunksQueried.withTags(TagSet.from(shardStats.tags))
+        PartLookupResult(shardNum, chunkMethod, partIds, Some(RecordSchema.schemaID(partition)),
+          queriedChunksCounter = chunksQueriedMetric)
       case MultiPartitionScan(partKeys, _)   =>
         val partIds = debox.Buffer.empty[Int]
         partKeys.flatMap(getPartition).foreach(p => partIds += p.partID)
-        PartLookupResult(shardNum, chunkMethod, partIds, partKeys.headOption.map(RecordSchema.schemaID))
+        val chunksQueriedMetric = shardStats.chunksQueried.withTags(TagSet.from(shardStats.tags))
+        PartLookupResult(shardNum, chunkMethod, partIds, partKeys.headOption.map(RecordSchema.schemaID),
+          queriedChunksCounter = chunksQueriedMetric)
       case FilteredPartitionScan(_, filters) =>
+        val chunksQueriedMetric = if (shardKeyLevelQueryMetricsEnabled) {
+          val metricTags = metricShardKeys.map { col =>
+            filters.collectFirst {
+              case ColumnFilter(c, Filter.Equals(filtVal: String)) if c == col => "metric" + col -> filtVal
+            }.getOrElse(col -> "unknown")
+          }.toMap
+          shardStats.chunksQueried.withTags(TagSet.from(metricTags))
+        } else shardStats.chunksQueried.withTags(TagSet.from(shardStats.tags))
```

Comment:
Not sure whether we really need to capture this.

Reply:
Downstream code needs a metric hence providing a default.